src/py/flwr/server/compat/legacy_context.py

```diff
@@ -35,9 +35,12 @@ class LegacyContext(Context):
     client_manager: ClientManager
     history: History
 
-    def __init__(
+    def __init__(  # pylint: disable=too-many-arguments
         self,
-        state: RecordSet,
+        node_id: int = 0,
```

Comment:
Wouldn't it be better to avoid the default here and force the user to pass `node_id`?

Reply:
IMO, we should require users to pass a `Context` object here. The arguments like `node_id` shouldn't be set by users; instead, they should be obtained automatically from an existing `Context` object. Wdyt? @danieljanes @jafermarq

Reply:
Yes, fully agree @panh99!